src/Aquifer.Public.API/Endpoints/Bibles/Alignments/Greek/Endpoint.cs

```diff
@@ -0,0 +1,275 @@
+        var upperBounds = BibleWordIdentifier.GetUpperBoundOfWord(bookCodeEnum, request.EndChapter, request.EndVerse, request.EndWord).WordIdentifier;
+
+        var bookData = await dbContext.BibleBooks
+            .Where(bb =>
+                bb.Bible.Enabled &&
+                !bb.Bible.RestrictedLicense &&
+                bb.Bible.GreekAlignment &&
+                bb.Bible.Id == request.BibleId &&
+                bb.Code == request.BookCode.ToUpper())
+            .Select(bb => new
+            {
+                BibleId = bb.Bible.Id,
+                BibleName = bb.Bible.Name,
+                BibleAbbreviation = bb.Bible.Abbreviation,
+                BookCode = bb.Code,
+                BookName = bb.LocalizedName,
+            })
+            .FirstOrDefaultAsync(ct);
+
+        if (bookData == null)
+        {
+            await SendNotFoundAsync(ct);
+            return;
+        }
+
+        var greekAlignmentSource = await dbContext.Database.GetDbConnection()
+            .QueryFirstOrDefaultAsync<string>(new CommandDefinition(GreekAlignmentSourceQuery, new { bibleId = request.BibleId }, cancellationToken: ct));
+
+        if (greekAlignmentSource == null)
+        {
+            await SendNotFoundAsync(ct);
+            return;
+        }
+
+        var bibleText = (await dbContext.Database.GetDbConnection()
+                .QueryAsync<BibleTextWithGreekAlignmentForeignKeysResult>(
+                    new CommandDefinition(
+                        BibleTextWithGreekAlignmentForeignKeysQuery,
+                        new
+                        {
+                            lowerBounds,
+                            upperBounds,
+                            bibleId = request.BibleId,
+                        },
+                        cancellationToken: ct)))
+            .ToList();
+
+        List<GreekWordResult> greekWordResults = [];
+        foreach (var batch in bibleText.Select(r => r.GreekWordId).Distinct().Order().Chunk(size: 2000))
+        {
+            greekWordResults.AddRange(await dbContext.Database.GetDbConnection()
+                .QueryAsync<GreekWordResult>(new CommandDefinition(GreekWordsQuery, new { greekWordIds = batch }, cancellationToken: ct)));
+        }
+
+        var greekWordResultById = greekWordResults.ToDictionary(gwr => gwr.Id);
+
+        List<GreekSenseResult> greekSenseResults = [];
+        if (request.ShouldReturnSenseData)
+        {
+            foreach (var batch in bibleText.Select(r => r.GreekSenseId).Distinct().Order().Chunk(size: 2000))
+            {
+                greekSenseResults.AddRange(await dbContext.Database.GetDbConnection()
+                    .QueryAsync<GreekSenseResult>(new CommandDefinition(GreekSensesQuery, new { greekSenseIds = batch }, cancellationToken: ct)));
+            }
+        }
+
+        var greekSenseResultById = greekSenseResults.ToLookup(r => r.Id);
+
+        var response = new Response
```

Comment:
I think this whole thing is really hard to read. I would break this into multiple private methods to clearly state what each part is building, and then that will also help with the excessive nesting.